snapcraft/__init__.py

```diff
@@ -288,8 +289,8 @@ def env(self, root):
         """
         return []
 
-    def set_target_machine(self):
-        """Set the target compilation architecture to machine."""
```

Comment:
Can we leave this function, continue taking the `machine` parameter, deprecate it, and simply have it call `enable_cross_compilation()`?